src/chain_sync/sync_status.rs

```diff
@@ -0,0 +1,230 @@
+}
+
+lotus_json_with_self!(SyncStatusReport);
+
+impl SyncStatusReport {
+    pub(crate) fn init() -> Self {
+        Self {
+            node_start_time: Utc::now(),
+            ..Default::default()
+        }
+    }
+
+    pub(crate) fn set_current_chain_head_key(&mut self, tipset_key: TipsetKey) {
+        self.current_head_key = Some(tipset_key);
+    }
+
+    pub(crate) fn set_current_chain_head_epoch(&mut self, epoch: ChainEpoch) {
+        self.current_head_epoch = epoch;
+    }
+
+    pub(crate) fn get_current_chain_head_epoch(&self) -> ChainEpoch {
+        self.current_head_epoch
+    }
+
+    pub(crate) fn set_network_head(&mut self, epoch: ChainEpoch) {
+        self.network_head_epoch = epoch;
+    }
+
+    pub(crate) fn get_network_head_epoch(&self) -> ChainEpoch {
+        self.network_head_epoch
+    }
+
+    pub(crate) fn get_current_chain_head_key(&self) -> Option<&TipsetKey> {
+        self.current_head_key.as_ref()
+    }
+
+    pub(crate) fn set_epochs_behind(&mut self, epochs_behind: i64) {
+        self.epochs_behind = epochs_behind;
+    }
+
+    pub(crate) fn get_epochs_behind(&self) -> i64 {
+        self.epochs_behind
+    }
+
+    pub(crate) fn set_status(&mut self, status: NodeSyncStatus) {
+        self.status = status;
+    }
+
+    pub(crate) fn get_status(&self) -> NodeSyncStatus {
+        self.status.clone()
+    }
+
+    pub(crate) fn update_active_forks(&mut self, active_forks: Vec<ForkSyncInfo>) {
+        self.active_forks = active_forks;
+    }
+
+    pub(crate) fn get_active_forks(&self) -> &Vec<ForkSyncInfo> {
+        &self.active_forks
+    }
+
+    pub fn get_last_updated(&self) -> &DateTime<Utc> {
+        &self.last_updated
+    }
+
+    pub(crate) fn update<DB: Blockstore + Sync + Send + 'static>(
+        &mut self,
+        state_manager: &Arc<StateManager<DB>>,
+        current_active_forks: Vec<ForkSyncInfo>,
+        stateless_mode: bool,
+    ) {
+        let heaviest = state_manager.chain_store().heaviest_tipset();
+        let current_chain_head_epoch = heaviest.epoch();
+        self.set_current_chain_head_key(heaviest.key().clone());
+        self.set_current_chain_head_epoch(current_chain_head_epoch);
+        let network_head_epoch = calculate_expected_epoch(
+            Utc::now().timestamp() as u64,
+            state_manager.chain_store().genesis_block_header().timestamp,
+            state_manager.chain_config().block_delay_secs,
+        );
+
+        self.set_network_head(network_head_epoch as ChainEpoch);
+        self.set_epochs_behind(network_head_epoch as i64 - current_chain_head_epoch);
+        let seconds_per_epoch = state_manager.chain_config().block_delay_secs;
+        let time_diff = (Utc::now().timestamp() as u64).saturating_sub(heaviest.min_timestamp());
+
+        match stateless_mode {
+            true => self.set_status(NodeSyncStatus::Offline),
+            false => {
+                if time_diff < seconds_per_epoch as u64 * 5 {
```

Comment:
worth commenting why `5`, best if we can avoid hardcoding

Reply:
made it a constant.

Reply:
@LesnyRumcajs any suggestions for this value here